Library/Homebrew/dev-cmd/prof.rb

```diff
@@ -27,7 +27,15 @@ def prof
     brew_rb = (HOMEBREW_LIBRARY_PATH/"brew.rb").resolved_path
     FileUtils.mkdir_p "prof"
     cmd = args.named.first
-    raise UsageError, "#{cmd} is a Bash command!" if Commands.path(cmd).extname == ".sh"
+
+    case Commands.path(cmd)&.extname
+    when ".rb"
+      # expected file extension so we do nothing
+    when ".sh"
+      raise UsageError, "`#{cmd}` is a Bash command!"
```

Comment:
```suggestion
      raise UsageError, "As `#{cmd}` is a Bash command it cannot be profiled with `brew prof`! Try `hyperfine` for benchmarking instead."
```

Reply:
I like this idea.

Reply:
```
/u/l/Homebrew (improve-prof-errors|✚1) [1]$ brew prof --version
Usage: brew prof [--stackprof] command [...]

Run Homebrew with a Ruby profiler. For example, brew prof readall.

      --stackprof                  Use stackprof instead of ruby-prof (the
                                   default).
  -d, --debug                      Display any debugging information.
  -q, --quiet                      Make some output more quiet.
  -v, --verbose                    Make some output more verbose.
  -h, --help                       Show this message.

Error: Invalid usage: `--version` is a Bash command!
Try `hyperfine` for benchmarking instead.
```

Wary of the 80 character limit I opted for this instead.